snmplib/snmp_api.c

```diff
@@ -6064,8 +6064,10 @@ _sess_read_dgram_packet(void *sessp, netsnmp_large_fd_set * fdset,
                                               SNMP_MAX_RCV_MSG_SIZE,
                                               &rcvp->opaque, &rcvp->olength);
     if (rcvp->packet_len == -1) {
-        sp->s_snmp_errno = SNMPERR_BAD_RECVFROM;
-        sp->s_errno = errno;
+        if (!sp) {
```

Comment:
It looks like an redundant check. If sp is NULL, then the execution is terminated early.